pipeline/hocon_func.go

```diff
@@ -1,12 +1,281 @@
 package pipeline
 
-import "fmt"
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	gohocon "github.com/goreflect/go_hocon"
+	"github.com/goreflect/gostructor/tags"
+)
 
 type HoconConfig struct {
+	fileName            string
+	configureFileParsed *gohocon.Config
+}
+
+func (config HoconConfig) getElementName(context *structContext) string {
+	currentTagHoconValue := context.StructField.Tag.Get(tags.TagHocon)
+	if strings.Contains(context.Prefix, currentTagHoconValue) {
+		fmt.Println("[HOCON]: Level: debug. Current field name: ", context.Prefix)
+		return context.Prefix
+	}
+	returnName := context.Prefix + "."
+	if currentTagHoconValue == "" {
+		returnName += context.StructField.Name
+	}
+
+	returnName += currentTagHoconValue
+	fmt.Println("[HOCON]: Level: debug. Current field name: ", returnName)
+	return returnName
+}
+
+func (config HoconConfig) GetComplexType(context *structContext) GoStructorValue {
+	loaded, structValue := config.typeSafeLoadConfigFile(context)
+	if !loaded {
+		return *structValue
+	}
+	valueIndirect := reflect.Indirect(context.Value)
+	switch valueIndirect.Kind() {
+	case reflect.Array:
+		return config.getSliceFromHocon(context)
+	case reflect.Slice:
+		return config.getSliceFromHocon(context)
+	case reflect.Map:
+		return config.getMapFromHocon(context)
+	default:
+		return config.GetBaseType(context)
+	}
+}
+
+// return true - if loaded config or successfully load config by filename
+func (config *HoconConfig) typeSafeLoadConfigFile(context *structContext) (bool, *GoStructorValue) {
+	if config.configureFileParsed == nil {
+		configParsed, err := gohocon.LoadConfig(config.fileName)
+		if err != nil {
+			return false, &GoStructorValue{notAValue: &NotAValue{
+				ValueAddress: context.Value.Interface(),
+				Error:        err,
+			},
+			}
+		}
+		config.configureFileParsed = configParsed
+		return true, nil
+	}
+	return true, nil
+}
+
+func (config *HoconConfig) GetBaseType(context *structContext) GoStructorValue {
+	configLoad, structValue := config.typeSafeLoadConfigFile(context)
+	if !configLoad {
+		return *structValue
+	}
+	valueIndirect := reflect.Indirect(context.Value)
+	path := config.getElementName(context)
+	switch valueIndirect.Kind() {
+	case reflect.Int:
+		loadValue, errLoading := config.configureFileParsed.GetString(path)
+		if errLoading != nil {
+			return NewGoStructorNoValue(context.Value.Interface(), errLoading)
+		}
+		valueParsedFromConfigFile, errParsed := strconv.ParseInt(loadValue, 10, 16)
```

Comment:
Why do you use 16 not 32 or 64 bits here?

Reply:
You've changed it to 32 but for x64 systems it will be 64, won't it?

Reply:
i changed this way by using parse into 64 int and passed by int interface cast into fields which type is int